stubs/bleach/bleach/linkifier.pyi

```diff
@@ -28,21 +21,34 @@ class Linker:
     def __init__(
         self,
         callbacks: Iterable[_Callback] = ...,
-        skip_tags: Container[str] | None = ...,
-        parse_email: bool = ...,
+        skip_tags: Container[str] | None = None,
+        parse_email: bool = False,
         url_re: Pattern[str] = ...,
         email_re: Pattern[str] = ...,
         recognized_tags: Container[str] | None = ...,
     ) -> None: ...
     def linkify(self, text: str) -> str: ...
 
 class LinkifyFilter(Filter):
-    callbacks: Any
+    callbacks: Iterable[_Callback]
     skip_tags: Container[str]
     parse_email: bool
-    url_re: Any
-    email_re: Any
+    url_re: Pattern[str]
+    email_re: Pattern[str]
     def __init__(
-        self, source, callbacks=..., skip_tags: Container[str] | None = ..., parse_email: bool = ..., url_re=..., email_re=...
+        self,
+        source,
+        callbacks: Iterable[_Callback] | None = ...,
+        skip_tags: Container[str] | None = None,
```

Comment:
`callbacks` and `skip_tags` need to be sized in order for [these lines](https://github.com/mozilla/bleach/blob/13d6c0c91d2b1bef52ad2b5a2812f76dda18298f/bleach/linkifier.py#L238-L239) to work (otherwise there's a chance that the iterable will be immediately consumed). So I think `Collection` would be better for both of these:

```suggestion
        callbacks: Collection[_Callback] | None = ...,
        skip_tags: Collection[str] | None = None,
```

(The same applies for the attribute annotations immediately above)

Reply:
A simple `if iter:` shouldn't consume an iterable. In the general case it should be equivalent with `if iter is not None:`.

Reply:
Good point, my reasoning about the iterable possibly being consumed was bogus. I still think a sized iterable is expected here, though. Unsized iterables are always truthy, even if they're empty:

```pycon
>>> x = iter([])
>>> bool(x)
True
```

Reply:
Actually maybe you're right and they're just guarding against `None` being passed in. Feel free to ignore me :)